### ISSUE TITLE
Adding >1 new line within a codeblock doesn't correctly match the indentation

### ISSUE THREAD

**Issue by aelsayed95** [python/cpython#119356]
# Bug report

### Bug description:

Adding more than one new line within a codeblock doesn't correctly match the indentation the previous line's indentation:

For example if I start with the following codeblock:
```python
>>> def f():
...     
...     a = 1
>>>
```

and modify it to add two additional new lines in-block, I end up with the following:

```python
>>> def f():
...     
... 
... 
...     a = 1
...     
>>>
```
Note that now, lines 2,5 have indentation, but lines 3 and 4 have no indentation.
This doesn't match IPython's behaviour.

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

macOS

**Comment by savannahostrowski** [comment]
In case it's helpful, this is also reproducible on Fedora, using the Devcontainer configuration in the repo.

**Comment by lysnikolaou** [comment]
Thanks for the issue @aelsayed95 and the confirmation @savannahostrowski!

This was fixed in #119606. Feel free to reopen in case you encounter this again.
